.github/workflows/yocto-build.yml

```diff
@@ -0,0 +1,411 @@
+
+      # the directory is required even if we don't mount the NFS share
+      - name: Create shared cache mount point
+        run: |
+          sudo mkdir -p "${YOCTO_CACHE_DIR}/$(whoami)"
+          sudo chown -R $(id -u):$(id -g) "${YOCTO_CACHE_DIR}"
+
+      - name: Mount shared NFS cache
+        if: env.YOCTO_CACHE_HOST != '' && contains(fromJSON(inputs.runs-on), 'self-hosted')
+        run: |
+          sudo mount -t nfs "${YOCTO_CACHE_HOST}:/" "${YOCTO_CACHE_DIR}" -o fsc,nolock
+          ls -al "${YOCTO_CACHE_DIR}"/$(whoami)
+
+      - name: Build
+        id: build
+        env:
+          HELPER_IMAGE_REPO: ghcr.io/balena-os/balena-yocto-scripts
+        run: |
+          ./balena-yocto-scripts/build/balena-build.sh \
+            -d "${MACHINE}" \
+            -t "${{ secrets.BALENA_API_KEY }}" \
+            -s "${YOCTO_CACHE_DIR}/$(whoami)" \
+            -g "${BARYS_ARGUMENTS_VAR}"
+
+          if grep -R "ERROR: " build/tmp/log/*; then
+            exit 1
+          fi
+
+          # if ! grep -R "Build for ${MACHINE} succeeded" build/tmp/log/*; then
+          #   exit 1
+          # fi
+
+          #du -cksh shared/yocto-cache/$(whoami)/*
+
+      # TODO: pre-install on self-hosted-runners
+      - name: Install zip packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y zip gzip
+
+      - name: Export prepare artifacts deploy path to env
+        env: 
+          DEVICE_TYPE_SLUG: ${{ steps.balena-lib.outputs.device_slug }}
+          VERSION: ${{ steps.balena-lib.outputs.os_version }}
+        run: | 
+          echo "DEPLOY_PATH=${{ runner.temp }}/deploy/${DEVICE_TYPE_SLUG}/${VERSION}" >> $GITHUB_ENV
+
+      # TODO: prepare artifacts manually to replace balena_deploy_artifacts
+      - name: Prepare artifacts
+        run: |
+          # DEBUG: check deploy path
+          echo "DEPLOY_PATH = ${DEPLOY_PATH}"
+
+          source "${automation_dir}/include/balena-deploy.inc"
+          balena_deploy_artifacts "${{ inputs.machine }}" "${DEPLOY_PATH}" false
+
+          # create links required for s3 prepare.ts
+          # mkdir -p "${DEPLOY_PATH}/${{ steps.balena-lib.outputs.device_slug }}"
+
+          # symlinks don't seem to work in the volume we mount to the helper image
+          #ln -sf ../.. "${DEPLOY_PATH}/${{ steps.balena-lib.outputs.device_slug }}/${{ steps.balena-lib.outputs.os_version }}"
+          
+          find "${DEPLOY_PATH}" -exec ls -lh {} \;
+
+          ls ${DEPLOY_PATH}
+
+          ls ${DEPLOY_PATH}/image
+
+          # Move the generated balena.yml into the deploy folder for easier access in deploy job
+          mv ${WORKSPACE}/balena.yml ${DEPLOY_PATH}
+
+      
+      #TODO:  select correct api key (balena machine or prod)
+
+      - name : Prepare Leviathan inputs
+        run: | 
+          # create "workspace" directory
+          echo LEVIATHAN_WORKSPACE="${WORKSPACE}/leviathan-workspace" >> $GITHUB_ENV
```

Comment:
@vipulgupta2048 

This technique doesn't allow using the env var in the same step:
"You can make an environment variable available to any subsequent steps in a workflow job by defining or updating the environment variable and writing this to the GITHUB_ENV environment file. The step that creates or updates the environment variable does not have access to the new value, but all subsequent steps in a job will have access."
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable